packages/app/src/debug/DebugSpec.vue

```diff
@@ -78,73 +122,140 @@
+  instance: CloudRunInstance
 }
 
 import { computed } from 'vue'
-import { IconActionRefresh } from '@cypress-design/vue-icon'
+import { IconActionRefresh, IconDocumentText } from '@cypress-design/vue-icon'
+import type { SpecDataAggregate, CloudRunInstance, CloudRunGroup } from '@packages/data-context/src/gen/graphcache-config.gen'
 import DebugFailedTest from './DebugFailedTest.vue'
+import StatsMetaData from './StatsMetadata.vue'
+import ResultCounts from '@packages/frontend-shared/src/components/ResultCounts.vue'
 import Button from '@packages/frontend-shared/src/components/Button.vue'
 import Tooltip from '@packages/frontend-shared/src/components/Tooltip.vue'
 import SpecNameDisplay from '../specs/SpecNameDisplay.vue'
 import { useI18n } from '@cy/i18n'
+import { useDurationFormat } from '../composables/useDurationFormat'
 import { posixify } from '../paths'
 import type { TestingTypeEnum } from '../generated/graphql'
 
 const { t } = useI18n()
 
 const props = defineProps<{
   spec: Spec
-  testResults: TestResults[]
-  foundLocally: boolean
+  testResults: {[thumbprint: string]: TestResults[]}
+  groups: {[groupId: string]: CloudRunGroup }
   testingType: TestingTypeEnum
+  isDisabled?: boolean
+  foundLocally: boolean
   matchesCurrentTestingType: boolean
 }>()
 
 const emits = defineEmits<{
   (event: 'switchTestingType', value: TestingTypeEnum): void
 }>()
 
+// helper function for formatting the number of passed, failed, and pending tests
+const debugResultsCalc = (min: number, max: number, specDuration: boolean = false) => {
+  if (min === max) {
+    return specDuration ? useDurationFormat(min ?? 0).value : min
+  }
+
+  return specDuration ? `${useDurationFormat(min ?? 0).value}-${useDurationFormat(max ?? 0).value}` : `${min}-${max}`
+}
+
 const specData = computed(() => {
+  const testsPassed = props.spec.testsPassed ?? 0
+  const testsFailed = props.spec.testsFailed ?? 0
+  const testsPending = props.spec.testsPending ?? 0
+  const specDuration = props.spec.specDuration ?? 0
+
   return {
-    path: props.spec.path,
+    path: `${props.spec.path}/`,
     fileName: props.spec.fileName,
     fileExtension: props.spec.fileExtension,
     failedTests: props.testResults,
+    testsPassed: debugResultsCalc(testsPassed.min ?? 0, testsPassed.max ?? 0),
+    testsFailed: debugResultsCalc(testsFailed.min ?? 0, testsFailed.max ?? 0),
+    testsPending: debugResultsCalc(testsPending.min ?? 0, testsPending.max ?? 0),
+    specDuration: debugResultsCalc(specDuration.min ?? 0, specDuration.max ?? 0, true),
+    groups: props.groups,
+    testingType: props.testingType,
     fullPath: props.spec.fullPath,
   }
 })
 
+const uniqueGrpIds: Set<string> = new Set()
+
+/**
+ * Helper function that maps each test's thumprint to all the groups in it
+ * return type: {[thumbprint: string]: CloudRunGroup[]}
+ */
+const groupsPerTest = computed(() => {
+  return Object.keys(props.testResults).reduce((acc, currThumbprint) => {
```

Comment:
You can type it here

```suggestion
  return Object.keys(props.testResults).reduce<Record<string, CloudRunGroup[]>>((acc, currThumbprint) => {
```